besu/src/main/java/org/hyperledger/besu/services/RpcEndpointServiceImpl.java

```diff
@@ -17,25 +17,42 @@
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Preconditions.checkNotNull;
 
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequest;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
 import org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.JsonRpcMethod;
 import org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.PluginJsonRpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
 import org.hyperledger.besu.plugin.services.RpcEndpointService;
 import org.hyperledger.besu.plugin.services.rpc.PluginRpcRequest;
+import org.hyperledger.besu.plugin.services.rpc.PluginRpcResponse;
+import org.hyperledger.besu.plugin.services.rpc.RpcResponseType;
 
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.Locale;
 import java.util.Map;
+import java.util.NoSuchElementException;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
 /** The RPC endpoint service implementation. */
 public class RpcEndpointServiceImpl implements RpcEndpointService {
   private final Map<String, Function<PluginRpcRequest, ?>> rpcMethods = new HashMap<>();
+  private Map<String, JsonRpcMethod> inProcessRpcMethods;
```

Comment:
Is this compatible with RPCs added via plugins as well, as per https://github.com/hyperledger/besu/pull/2754? 

I'm not 100% sure if this scenario makes sense, but perhaps we'd want to call a plugin's RPC method via another plugin, i.e.
* plugin1 registers rpc1 with besu
* plugin2 calls rpc1 in-process

Reply:
This a quick test, and found that it is possible to call a method defined in another plugin, but only if the plugin namespace is enabled via `rpc-http-api`, that's not optimal, since should be enough to enable it only using `Xin-process-rpc-apis`, will improve that on a following PR with proper tests